pints/_mcmc/_relativistic.py

```diff
@@ -0,0 +1,437 @@
+
+    def scaled_epsilon(self):
+        """
+        Returns scaled epsilon used in leapfrog algorithm
+        """
+        return self._scaled_epsilon
+
+    def _set_scaled_epsilon(self):
+        """
+        Rescales epsilon along the dimensions of step_size
+        """
+        self._scaled_epsilon = np.zeros(self._n_parameters)
+        for i in range(self._n_parameters):
+            self._scaled_epsilon[i] = self._epsilon * self._step_size[i]
+
+    def set_epsilon(self, epsilon):
+        """
+        Sets epsilon for the leapfrog algorithm
+        """
+        epsilon = float(epsilon)
+        if epsilon <= 0:
+            raise ValueError('epsilon must be positive for leapfrog algorithm')
+        self._epsilon = epsilon
+        self._set_scaled_epsilon()
+
+    def set_hamiltonian_threshold(self, hamiltonian_threshold):
+        """
+        Sets threshold difference in Hamiltonian value from one iteration to
+        next which determines whether an iteration is divergent.
+        """
+        if hamiltonian_threshold < 0:
+            raise ValueError('Threshold for divergent iterations must be ' +
+                             'non-negative.')
+        self._hamiltonian_threshold = hamiltonian_threshold
+
+    def set_hyper_parameters(self, x):
+        """
+        The hyper-parameter vector is ``[leapfrog_steps, leapfrog_step_size,
+        mass, c]``.
+
+        See :meth:`TunableMethod.set_hyper_parameters()`.
+        """
+        self.set_leapfrog_steps(x[0])
+        self.set_leapfrog_step_size(x[1])
+        self.set_mass(x[2])
+        self.set_speed_light(x[3])
+
+    def set_leapfrog_steps(self, steps):
+        """
+        Sets the number of leapfrog steps to carry out for each iteration.
+        """
+        steps = int(steps)
+        if steps < 1:
+            raise ValueError('Number of steps must exceed 0.')
+        self._n_frog_iterations = steps
+
+    def set_leapfrog_step_size(self, step_size):
+        """
+        Sets the step size for the leapfrog algorithm.
+        """
+        a = np.atleast_1d(step_size)
+        if len(a[a < 0]) > 0:
+            raise ValueError(
+                'Step size for leapfrog algorithm must' +
+                'be greater than zero.'
+            )
+        if len(a) == 1:
+            step_size = np.repeat(step_size, self._n_parameters)
+        elif not len(step_size) == self._n_parameters:
+            raise ValueError(
+                'Step size should either be of length 1 or equal to the' +
+                'number of parameters'
+            )
+        self._step_size = step_size
+        self._set_scaled_epsilon()
+
+    def set_mass(self, mass):
+        """ Sets scalar mass. """
+        if isinstance(mass, list):
+            raise ValueError('Mass must be scalar.')
+        if mass <= 0:
+            raise ValueError('Mass must be positive.')
+        self._mass = mass
+
+    def set_speed_light(self, c):
```

Comment:
Uhm... Maybe some quotes around "speed of light" in the docstring?

Reply:
Haha, yep. I don't want to be playing God. Done.